### DISCUSSION TITLE
syncing an O365 SharePoint Shared Library, files deleted in the library are not deleted on the client... sort of...

### DISCUSSION THREAD

**Discussion by cody0922** [abraunegg/onedrive#1627] [Q&A] [answered]
Ubuntu 20.04 (on AWS) 
onedrive v2.4.13-1~np1  (was previously 2.4.10 when issue was noticed)

onedrive is configured to download-only 

when we first set this up a few months ago, i'm pretty sure i tested deletes in the library being synced on the client, but i can't say with 100% certainty.

even running --resync there were files on the client that did not exist in the library

to test further, i created test files in a couple of different folders through the sharepoint web and manually ran --synchronize; the files were synced as expected.  i then deleted those files through sharepoint web and ran another sync. the files were not removed by the client. 

i then noticed we were on an unsupported version and that there were several fixes related to one-way syncs in more recent versions.  updated to 2.4.13.

now when i create the test files through sharepoint web, sync, delete the files, and sync again, those changes are being reflected exactly how i would expect..  so that's great!

however, there are still orphaned files, from before the onedrive client update, that still exist on the client but not in the library. even with a --resync the client is not removing those orphaned files.

i'm missing something, but can't figure out what it is.

**Comment by abraunegg** [comment, marked as answer]
@cody0922 
> i then noticed we were on an unsupported version and that there were several fixes related to one-way syncs in more recent versions. updated to 2.4.13.
> ...
> however, there are still orphaned files, from before the onedrive client update, that still exist on the client but not in the library. even with a --resync the client is not removing those orphaned files.
>
> i'm missing something, but can't figure out what it is.

The issue is compounded by you using `--download-only` and `--resync` and using originally an unsupported client version.

`--resync` will remove the local cache database, thus, there is no record of what local files existed, what is the correct true nature of what is required. Essentially dont use `--resync` unless you make a configuration change and the application directs you to do so.

Because you are doing `--download-only`, there is no way to then say - anything that I dont download, delete locally - as - that may lead to a potential data loss situation.

The only way to resolve this is to remove your unwanted | orphaned local files manually, ensure you are using the correct application version & dont use `--resync` unless the application tells you to.

**Reply by cody0922** [reply]
originally we weren't using --resync, just --monitor via service (and download-only=true via config).. only tried --resync when we discovered the files weren't being removed, and that was prior to updating to 2.4.13

so then chances are, due to i suppose a bug the previous version(?) we were using, deletes weren't being sync'ed to begin with?

since there's no way for us to know where all of the orphaned files are, i think the best route would then be to blow everything away on the client and redownload everything.

what's the process for that?

**Reply by cody0922** [reply]
btw, the version we were using was supported when we first starting using it

**Comment by abraunegg** [comment]
@cody0922 
> since there's no way for us to know where all of the orphaned files are, i think the best route would then be to blow everything away on the client and redownload everything.
>
> what's the process for that?

As you are using Ubuntu 20.04 & Ubuntu packages, you first need to ensure you have read and understand the known issues with using Ubuntu packages: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages

If you can .. upgrade your Ubuntu to 21.x to avoid the Ubuntu 20.x core dump issue, or potentially stop using Ubuntu and using something that is not Ubuntu | Ubuntu based.

Second you *must* ensure you remove the symbolic link as per the first issue. This is super critical. Once you have done this, ensure that any systemd service for 'onedrive' is disabled, stopped and that there is no 'onedrive' process running. It may even be worth restarting the system to ensure it comes up clean with no 'onedrive' process active.

With the system in this 'clean' state, the best route would be to remove all files | folders under your configured sync_dir path - typically this will be `~/OneDrive` - but if you have modified your configuration, then you will need to remove the files from the right path.

Once this is done, you have an option:
* Use `--resync` 
* Manually remove the `items.sqlite3` file from your configuration directory - typically `~/.config/onedrive/`

Sync your data from OneDrive -> Local, test first with `--dry-run`:
* Use: `onedrive --synchronize --download-only --dry-run {--resync optional based on action above}`

Once you are happy that this will perform the action you want, run without `--dry-run`
* Use: `onedrive --synchronize --download-only {--resync optional based on action above}`

Confirm your data has been downloaded

Then, your actions | choice here:
1. Enable system service - which will use --monitor and upload | download data
2. Manually run the client as per requirement


> btw, the version we were using was supported when we first starting using it

This might be the case, but thanks to Ubuntu and their packaging process, the Ubuntu packages are just not maintained, and the PPA archive depends on the Debian packages, which themselves are out-of-date .. Suggest that you monitor | watch the GitHub repository so that you get notified when there is a new release.

**Reply by cody0922** [reply]
we are aware of the issues with ubuntu; so far we haven't seen the segfault / core-dump issue. and i already removed the symlink when this was originally set up; and double checked after upgrading that it was still not present.  this is running on an AWS instance, so I went with what i was familiar with. if the issues with ubuntu become more bothersome, i might just rebuild it on RHEL.

at any rate i was hoping it would be a "simple" matter of removing everything in the sync folder and doing a --resync (after stopping and disabling onedrive, of course).  thank you for explaining the process in detail.

**Reply by cody0922** [reply]
oh, i just checked systemctl  status.. we are seeing the core-dump
